src/window/src/lib.rs

```diff
@@ -88,53 +145,74 @@ pub trait AdvancedWindow: Window + Sized {
     fn set_exit_on_esc(&mut self, value: bool);
 
-    /// Sets whether to exit when pressing esc.
+    /// Sets whether to exit when pressing the Esc button.
+    /// 
+    /// This version moves the current window data, 
+    /// unlike [`set_exit_on_esc()`](#method.set_exit_on_esc), so 
+    /// that it can be used in method chaining.
     fn exit_on_esc(mut self, value: bool) -> Self {
         self.set_exit_on_esc(value);
         self
     }
 
-    /// Sets whether to capture/grab cursor.
+    /// Sets whether to capture/grab the cursor.
+    ///
     /// This is used to lock and hide cursor to the window,
     /// for example in a first-person shooter game.
     fn set_capture_cursor(&mut self, value: bool);
 
-    /// Sets whether to capture/grab cursor (see `set_capture_cursor`).
+    /// Sets whether to capture/grab the cursor.
+    ///
+    /// This version moves the current window data, 
+    /// unlike [`set_capture_cursor()`](#method.set_capture_cursor), so
+    /// that it can be used in method chaining.
     fn capture_cursor(mut self, value: bool) -> Self {
         self.set_capture_cursor(value);
         self
     }
 }
 
 /// Trait for OpenGL specific operations.
+///
+/// For backends that support OpenGL, this trait allows client code to
+/// access the raw OpenGL context.
 pub trait OpenGLWindow: Window {
-    /// Returns the address of an OpenGL function if it exist, else returns null pointer.
+    /// Returns the address of the specified OpenGL function if it exists.
+    ///
+    /// If the function does not exist, it returns a null pointer.
     fn get_proc_address(&mut self, proc_name: &str) -> ProcAddress;
 
-    /// Returns true if this context is the current context.
+    /// Returns true if this window's gl context is the current gl context.
     fn is_current(&self) -> bool;
 
-    /// Make this context current.
+    /// Make the window's gl context the current gl context.
     fn make_current(&mut self);
 }
 
-/// Settings for window behavior.
+/// Settings structure for window behavior.
+///
+/// It stores everything that needs to be customized when constructing most
+/// windows. This structure makes it easy to create multiple windows with the
+/// same settings, and it also makes piston's multiple backends easier to
+/// implement for piston devs.
 pub struct WindowSettings {
     /// Title of the window.
     title: String,
     /// The size of the window.
     size: Size,
-    /// Number samples per pixel (anti-aliasing).
+    /// Number samples per pixel.
+    ///
+    /// Used during anti-aliasing.
     samples: u8,
-    /// If true, the window is fullscreen.
+    /// If true, the window should be fullscreen.
     fullscreen: bool,
-    /// If true, exit when pressing Esc.
+    /// If true, the window should exit when pressing the Esc key.
     exit_on_esc: bool,
     /// If true, enable vsync.
     vsync: bool,
-    /// An optional OpenGL version.
+    /// An optional OpenGL instance.
```

Comment:
"version" was correct. It is an enum that tells the window backend which OpenGL version to use.